Sources/GDSAnalytics/README.md

```diff
@@ -150,6 +150,56 @@ let event = LinkEvent(textKey: "Confirm your identity another way",
 | link domain | The web domain of the link that was pressed. | Lowercase. Limited to 100 characters. | `https://signin.build.account.gov.uk` |
 | external | Indicates whether the link is to an external domain. | Lowercase. | `true` or `false` |
 
+### Popup Alerts
+
+This type of event should be logged when a call-to-action is displayed, usually this would be a `UIAlertController` but it may also be used for other modal inputs. Use the `PopupEvent` type for this.
+
+```swift
+let event = PopupEvent(textKey: "Change your permissions")
+```
+
+#### Expected Parameters:
+
+| Parameter | Description | Expected Format | Example |
+| ------------- |-------------| -----| --- |
+| name | The name of the event that occured, for popups this is `popup`. | N/A | `popup` |
+| type | The type of event that occured, for popups this is `call to action`. | N/A | `call to action` |
+| text | The text that is displayed on the popup. | Lowercase. Limited to 100 characters. | `change your permissions` |
+
+### Action Menu 
+
+This type of event fires when a custom systemAlert popup appears on the screen, usually this would be a `UIAlertController` of the type `.actionSheet`. Use the `ActionMenuEvent` type for this.
+
+```swift
+let event = ActionMenuEvent(textKey: "Change your permissions")
```

Comment:
The example we're using here seems similar to the one above, please can we try and find a good example for each from our existing (or planned) journey ?

Reply:
might even be worth adding screenshots of the examples

Reply:
I think this might include Sam, as I only have the info from the spec which I have put above. Wonder in the interest of time does it make sense to do this as a separate piece of work? Agree that it is very important to add this, for ease of use but may be a day or two by the time I can get time with him to get examples etc of where to use these. However, if we think it is essential to have this to merge this, I'm happy to pick that up :)

Reply:
Opened another ticket for this, I will see to it that we get that in asap but will merge this separate to that.